File: adsterox123-cs50-2017-x-pset6/caesar.py
import sys
if len(sys.argv) == 2:

    #declare variables

    k = int(sys.argv[1])

    #get user input
    plaintext = input("plaintext: ")
    print("ciphertext: ",end = '')
    #encrypt - iterate over each caracter in the string (plaintext)
    for i in range(len(plaintext)):

        #uppercase letters
        if plaintext[i].isupper():

            #shift plaintext by key
            ciphertext = ((ord(plaintext[i]) - 65) + k) % 26
            #print encrypted message character by character
            ciphertext = ciphertext + 65
            print((chr(ciphertext)),end = '')

        #lowercase letters
        elif plaintext[i].islower():

            #shift plaintext by key
            ciphertext = ((ord(plaintext[i]) - 97) + k) % 26;
            #print encrypted message character by character
            ciphertext = ciphertext + 97
            print((chr(ciphertext)),end = '')

        #if non-alphabet
        elif plaintext[i].isalpha() != True:
            #print without shifting/changing
            print(plaintext[i],end = '')



    #newline to make the output more neat and presentable
    print("");


#if there are more or less than two cammand line arguments
else:

    #error message
    print("Usage: ./caesar k")

File: adsterox123-cs50-2017-x-caesar/caesar.c
//caesar.c

//takes user input and encrypts it using a caesar cypher

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    //program works only if there are two arguments
    if (argc == 2)
    {
        //declare variables
        string key = argv[1];
        int k = atoi(key);
        char ciphertext;
        //get user input
        printf("plaintext: ");
        string plaintext = get_string();
        int i;
        printf("ciphertext: ");
        //encrypt - iterate over each caracter in the string (plaintext)
        for (i = 0; i < strlen(plaintext); i++)
        {
            //uppercase letters
            if (isupper(plaintext[i]))
            {
                //shift plaintext by key
                 ciphertext = ((plaintext[i] - 65) + k) % 26;
                 //print encrypted message character by character
                 printf("%c", ciphertext + 65);
            }
            //lowercase letters
            else if (islower(plaintext[i]))
            {
                //shift plaintext by key
                ciphertext = ((plaintext[i] - 97) + k) % 26;
                //print encrypted message character by character
                printf("%c", ciphertext + 97);
            }
            //if non-alphabet
            else
            {
                //print without shifting/changing 
                printf("%c", plaintext[i]);
            }
            
        }
        //newline to make the output more neat and presentable
        printf("\n");
        return 0;
    }
    //if there are more or less than two cammand line arguments
    else
    {
        //error message
        printf("Usage: ./caesar k\n");
        return 1;
    }
}